api/src/extensions/lib/get-extensions.ts

```diff
@@ -1,54 +1,25 @@
 import { useEnv } from '@directus/env';
-import type { Extension } from '@directus/extensions';
-import { getLocalExtensions, getPackageExtensions, resolvePackageExtensions } from '@directus/extensions/node';
+import { resolveDependencyExtensions, resolveExtensions } from '@directus/extensions/node';
 import { useLogger } from '../../logger.js';
 import { getExtensionsPath } from './get-extensions-path.js';
 
 export const getExtensions = async () => {
 	const env = useEnv();
 	const logger = useLogger();
+	const extensions = await resolveExtensions(getExtensionsPath());
+	const extensionNames = extensions.map(({ name }) => name);
 
-	const loadedExtensions = new Map();
-	const duplicateExtensions: string[] = [];
+	/** Extensions that are listed in the root package.json dependencies */
+	const dependencyExtensions = await resolveDependencyExtensions(env['PACKAGE_FILE_LOCATION'] as string);
 
-	const filterDuplicates = (extension: Extension) => {
-		const isExistingExtension = loadedExtensions.has(extension.name);
-
-		if (isExistingExtension) {
-			duplicateExtensions.push(extension.name);
-			return;
-		}
-
-		if (extension.type === 'bundle') {
-			const bundleEntryNames = new Set();
-
-			for (const entry of extension.entries) {
-				if (bundleEntryNames.has(entry.name)) {
-					// Do not load entire bundle if it has duplicated entries
-					duplicateExtensions.push(extension.name);
-					return;
-				}
-
-				bundleEntryNames.add(entry.name);
-			}
+	const dedupedDependencyExtensions = dependencyExtensions.filter((ext) => {
+		if (extensionNames.includes(ext.name)) {
+			logger.warn(`Node module extension "${ext.name}" already exists within the extensions folder.`);
```

Comment:
Not sure, but would probably speak of a package extension
```suggestion
			logger.warn(`Package extension "${ext.name}" already exists within the local extensions folder.`);
```